### PR TITLE
small util for key management

### PR DESCRIPTION
**Pull request by paspo**
With this util, we add a way to generate and verify a keypair. The keypair generation is useful for first installation, especially when https://github.com/rustdesk/rustdesk-server/pull/61 gets approved.

The command will be something like:

```bash
docker run --rm --entrypoint /usr/bin/rustdesk-utils rustdesk/rustdesk-server:latest genkeypair
```

The expected output is something like this:

```text
Public Key:  e61C1tOULMqVbAf0kTlUbyBFNZ1eObXIOQ1O0BSSB/o=
Secret Key:  ZuDfy4g9/HO1U6003hTQSaFQNBT7XPpz4B4F2TOw9Uh7rULW05QsypVsB/SROVRvIEU1nV45tcg5DU7QFJIH+g==
```

The other available command is `validatekeypair` which is used to check if the keypair is valid:

```bash
rustdesk-utils validatekeypair 'e61C1tOULMqVbAf0kTlUbyBFNZ1eObXIOQ1O0BSSB/o=' 'ZuDfy4g9/HO1U6003hTQSaFQNBT7XPpz4B4F2TOw9Uh7rULW05QsypVsB/SROVRvIEU1nV45tcg5DU7QFJIH+g=='
```

This will be used at the container startup to check keypair validity before program startup.

# NOTE

This is the first time i use rust. I really can use the help of a skilled developer to check on this work; that's why this is a draft.

### CODE REVIEW

**Comment by rustdesk** [comment]
> a skilled developer

The goodness of of Rust is that a new developer can also write good code.

**Comment by rustdesk** [comment]
And your Rust code seems not formatted, you can run `cargo fmt`

**Comment by rustdesk** [comment]
@paspo have you seen my code review? I noticed some one can not see the in-code review.

image

**Review by rustdesk — CHANGES_REQUESTED** [review]

File: src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+use std::str;
+use std::env;
+use std::process;
+
+fn print_help() {
+    println!("Usage:");
+    println!("  rustdesk-util [command]\n");
+    println!("Available Commands:");
+    println!("  genkeypair                                   Generate a new keypair");
+    println!("  validatekeypair [public key] [secret key]    Validate an existing keypair");
```
[the remark this answers is not in the document]
Comment (paspo) [reply src/utils.rs:11]: ok

File: src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    println!("Secret Key:  {secret_key}");
+}
+
+fn validate_keypair(pk: &str, sk: &str) {
+    let sk1 = base64::decode(&sk);
+    match sk1 {
+        Ok(_) => {},
+        Err(_) => {
+            println!("Invalid secret key");
+            process::exit(0x0001);
```
[the remark this answers is not in the document]
Comment (paspo) [reply src/utils.rs:34]: I read [here](https://docs.rs/failure/0.1.1/failure/macro.bail.html) that this macro is used for returning something from a function. Here I want the program to quit, with an exit status code to differentiate from a correct exit.

File: src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    let sk1 = sk1.unwrap();
+
+    let secret_key = sign::SecretKey::from_slice(sk1.as_slice());
+    match secret_key {
+        Some(_) => {},
+        None => {
+            println!("Invalid Secret key");
+            process::exit(0x0001);
+        },
+    }
```
[the remark this answers is not in the document]
Comment (paspo) [reply src/utils.rs:46]: that's a lot better in readability

File: src/utils.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rustdesk) [reply src/utils.rs:34]: You can quit from the calling function.

[the remark this answers is not in the document]
Comment (paspo) [reply src/utils.rs:34]: ok.

Given this code:

```rust
use hbb_common::ResultType;

fn validate_keypair(pk: &str, sk: &str) -> ResultType<()>  {
    let sk1 = base64::decode(&sk);
    // ....
}
```

what is the "rust way" of saying "if decoding doesn't work, return with an error string"?
bonus points if you fit this in one line...

**Pull request merged**